### ISSUE TITLE
How to add new application type with its associated actions

### ISSUE THREAD

**Issue by mardukbp** [cerberustesting/cerberus-core#2407]
I developed a C# library of actions for automating a Windows application. The C# library exposes an XML-RPC interface for interaction with other languages. How can I incorporate these actions in Cerberus? In other words, if I implement an XML-RPC client in Java, how can I make the API calls available to Cerberus?

**Comment by vertigo17** [comment]
Not sure to understand ;-(.
I understand your library allows to interact with a windows application on the robot side ?
What type of application does it allow to interact with ?

If it allows to interact with any Windows application, that would be a nice Cerberus improvement and alternative to Sikuli.

To implement that,  we would need to either:
+ Enrich the cerberus-robot-extention module that execute on the robot side. That robot extension would expose your application actions in REST mode so that Cerberus can trigger the necessary actions.
+ If your application already expose a REST API, we would have cerberus to mapp all Cerberus actions to the necessay RESt API.

Do you have some documentation on your library and the feature it has ?

**Comment by mardukbp** [comment]
Now I realize I did a pretty bad job at explaining. Apologies.

The background story is that I developed a [library of keywords](https://imbus.github.io/robotframework-robosapiens/) for automating the SAP GUI using Robot Framework. I did it while working for a client and now my employer open sourced the code. The client wanted to build a web GUI to make the keywords available to domain experts and I suggested using Cerberus, instead of implementing something from scratch. But in the end the web GUI was built.

For automating Windows applications I have successfully used [FlaUI](https://github.com/FlaUI/FlaUI). Sometimes in combination with Sikuli.

My library no longer exposes the keywords via XML-RPC, but rather uses JSON-RPC over STDIN/STDOUT. And I currently have no need to use Cerberus. But I hope this issue leads to Cerberus having an extension interface for adding keyword libraries written in other programming languages.
